src/Zenith.sol

```diff
@@ -6,30 +6,25 @@ import {Passage} from "./Passage.sol";
 contract Zenith is Passage {
     /// @notice The address that is allowed to set/remove sequencers.
     address public immutable sequencerAdmin;
+    /// @notice The block number at which the Zenith contract was deployed.
+    uint256 public immutable deployBlockNumber;
```

Comment:
if it's an immutable it gets stored in a code region, reading would be slightly easier+cheaper if this were stored in state instead

Reply:
gotcha, was assuming it'd be read through RPC interface for some reason? changed: https://github.com/init4tech/zenith/pull/48/commits/d462e0905ca7dd9c5fd8aa97ff0092961d548576